app/code/Magento/ConfigurableProduct/Block/Cart/Item/Renderer/Configurable.php

```diff
@@ -66,7 +66,7 @@ public function getProductForThumbnail()
         if ($this->_scopeConfig->getValue(
             self::CONFIG_THUMBNAIL_SOURCE,
             \Magento\Store\Model\ScopeInterface::SCOPE_STORE
-        ) == ThumbnailSource::OPTION_USE_PARENT_IMAGE ||
+        ) == ThumbnailSource::OPTION_USE_PARENT_IMAGE || (empty($this->getChildProduct()) ||
             !($this->getChildProduct()->getThumbnail() && $this->getChildProduct()->getThumbnail() != 'no_selection')
```

Comment:
@orlangur We ran into this in production this morning, had a 500 error on /checkout/cart/ for a user and traced it back to this code. It was being called from the view/frontend/templates/cart/item/default.phtml template in the  checkout module.